src/Tribe/Views/V2/Rest_Endpoint.php

```diff
@@ -16,6 +38,106 @@ class Rest_Endpoint {
 	 * @return  string
 	 */
 	public function get_url() {
-		return get_rest_url( null, Service_Provider::NAME_SPACE . '/html' );
+		if ( ! $this->is_available() ) {
+			$url = admin_url( 'admin-ajax.php' );
+			return add_query_arg( [ 'action' => static::$ajax_action ], $url );
+		}
+
+		return get_rest_url( null, static::NAMESPACE . '/html' );
+	}
+
+	/**
+	 * Register the endpoint if available.
+	 *
+	 * @since  TBD
+	 *
+	 * @return boolean If we registered the endpoint.
+	 */
+	public function register() {
+		if ( ! $this->is_available() ) {
+			return false;
+		}
+
+		return register_rest_route( static::NAMESPACE, '/html', [
+			'methods'             => Server::READABLE,
+			/**
+			 * @todo  Make sure we do proper handling of cache longer then 12h.
+			 */
+			'permission_callback' => static function ( Request $request ) {
+				return wp_verify_nonce( $request->get_param( '_wpnonce' ), 'wp_rest' );
+			},
+			'callback' => static function ( Request $request ) {
+				View::make_for_rest( $request )->send_html();
+			},
+			'args' => [
+				'url' => [
+					'required'          => true,
+					'validate_callback' => static function ( $url ) {
+						return is_string( $url );
+					},
+					'sanitize_callback' => static function ( $url ) {
+						return filter_var( $url, FILTER_SANITIZE_URL );
+					}
+				],
+			],
+		] );
+	}
+
+	/**
+	 * When REST is not available add ajax fallback into the correct action.
+	 *
+	 * @since  TBD
+	 *
+	 * @return void
+	 */
+	public function maybe_enable_ajax_fallback() {
+		if ( $this->is_available() ) {
+			return;
+		}
+
+		$action = static::$ajax_action;
+		add_action( "wp_ajax_{$action}", [ $this, 'ajax_fallback' ] );
+		add_action( "wp_ajax_nopriv_{$action}", [ $this, 'ajax_fallback' ] );
+	}
+
+	/**
+	 * AJAX fallback for when REST endpoint is disabled. We try to mock a WP_REST_Request
+	 * and use the same method behind the scenes to make sure we have consistency.
+	 *
+	 * @since  TBD
+	 *
+	 * @return void
+	 */
+	public function ajax_fallback() {
+		$request = new Request( 'GET', static::NAMESPACE . '/html' );
+
+		$body_params = (array) $_GET;
```

Comment:
Sanitize here; as first step drop any argument we're not using, as a second step sanitize each by type.

Reply:
I am not sure if we want to add a layer of Sanitization here, because it should just pass the params directly without filtering since it would be a second place to check for more params to be used. I would like that to be done on the `make_for_rest`. Or do you mean something else?

Reply:
Had a convo with @lucatume will abstract that info and make sure we are running the same checks for both.